internal/engine/dockerprune/docker_pruner.go

```diff
@@ -0,0 +1,176 @@
+package dockerprune
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+
+	"github.com/windmilleng/tilt/internal/docker"
+	"github.com/windmilleng/tilt/internal/sliceutils"
+	"github.com/windmilleng/tilt/internal/store"
+	"github.com/windmilleng/tilt/pkg/logger"
+)
+
+// How often to prune Docker images while Tilt is running
+const dockerPruneInterval = time.Hour
+
+// Prune Docker objects older than this
+// TODO: configurable from Tiltfile for special cases
+const defaultMaxAge = time.Hour * 6
```

Comment:
Not blocking, but as discussed offline I worry we'll end up in cases where we're deleting the newest build of a resource when the user is using a remote k8s cluster.

Reply:
hopefully that'll be addressed by the "prune every X builds" setting coming in my next commit